### ISSUE TITLE
[pickers] New component: DateTimeRangePicker

### ISSUE THREAD

**Issue by rodrigofariow** [mui/mui-x#4547]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Not sure how this should be implemented according to the material design specs exactly but I assume you guys have a better and more conceivable idea than me. 😁

### Motivation 🔦

My company has recently bought the MUI X PRO license and despite it's clear benefit (specially `DataGridPro`) we were expecting a more complete `DateRangePicker`.
~~**Note that we are aware that it's still in the lab** but~~ I think there is a real need to have the option to select the time along with dates to formulate a date range.

A common use case is to define a period of time where some type of user events have occured. Be it activity reports from a user, be it the amount of money the user has spend during that exact period. There are endless possibilities.

Is this something that is on your roadmap? I haven't found any info regarding this question.

Thank you! :) 

### Benchmark

- Ant Design, for instance, has support for this -> 
![image](https://user-images.githubusercontent.com/22502084/145814399-7fc1de01-1f45-4851-a64a-7eb27b413f4c.png)
- More: https://mui-org.notion.site/Date-Time-Range-Picker-component-7a7cc24a3ba34ad0a04ebf6458eb5c44 (public) / https://www.notion.site/mui-org/Date-Time-Range-Picker-component-7a7cc24a3ba34ad0a04ebf6458eb5c44 (private)

### Other requests

- https://www.notion.so/mui-org/Sharechat-Prateek-Amar-2aee32660fb044df8bf0337f4f357feb#e09c842ed3da47409bfe7257e5b7dac5
- https://mui.zendesk.com/agent/tickets/7124

**Comment by fdcolongonzalez** [comment]
Is there an ETA for this new feature?

**Comment by hoa00** [comment]
upvote for this feature 👍

**Comment by lawcia** [comment]
👍 would also like this

**Comment by ltphuoc** [comment]
👍 upvote

**Comment by trunglvse151138** [comment]
would like too see this soon 🔥🔥

**Comment by CMeyer19** [comment]
Would love to have this feature

**Comment by adarsh-drishya** [comment]
please have this feature

**Comment by purefun** [comment]
👍 would also like this

**Comment by kingkong0828** [comment]
would like too see this soon

**Comment by Jfish-mastt** [comment]
👍  We would like this also

**Comment by piyushroongta** [comment]
would like  have this

**Comment by WinmezzZ** [comment]
+1

**Comment by desisiv** [comment]
👍 upvote

**Comment by filipstojakovic** [comment]
👍 upvote

**Comment by lindapr** [comment]
👍

**Comment by AbrarChowdhury** [comment]
need this!

**Comment by iragreenstein** [comment]
yes, please

**Comment by bannarisoftwares** [comment]
Need this........  Please merge this feature....

**Comment by wasikarn** [comment]
Would like to have this

**Comment by hubert77500** [comment]
please, this is very needed!

**Comment by bryantobing12** [comment]
would be nice if we have this for free :).

**Comment by jjmorse** [comment]
This would be great to have 👍

**Comment by naveenchittiphp** [comment]
Please make it fast lot of projects are waiting for it 👍.

**Comment by alvamanu** [comment]
How about the ability to add predefined ranges? "Last week", "Last month" etc. Also, typed in time inputs would be great.

**Comment by flaviendelangle** [comment]
> How about the ability to add predefined ranges? "Last week", "Last month" etc.

You can have a look at https://github.com/mui/mui-x/issues/4563

**Comment by heshamnaim** [comment]
Upvote!!!

**Comment by robinjhuang** [comment]
Upvote!!

**Comment by geraldchen890806** [comment]
Upvote!!

**Comment by dipali-oc** [comment]
Upvote

**Comment by zakka258** [comment]
upvote

**Comment by mlshvdv** [comment]
Upvote

**Comment by anabiadev** [comment]
Upvote !!!

**Comment by josempineiro** [comment]
Upvote!!

**Comment by afilp** [comment]
Upvote

**Comment by pablo-johnson** [comment]
Upvote!

**Comment by LuisGVV** [comment]
Definitely need this! Upvote!

**Comment by ten-skp** [comment]
Upvote

**Comment by flaviendelangle** [comment]
Small update here.
We know that a lot of people are waiting for this feature and it remains one of our top priority.

The team is currently rebuilding the input part of the pickers from scratch to provide a better experience.
We should be able to soon release a first documented (but still unstable) version of those components soon.
You can find a basic internal documentation of those components [here](https://next.mui.com/x/react-date-pickers/date-field) if you are curious.

The plan is then to create new keyboard-only components from *time range* and *date time range* in the coming weeks.
And then to build new pickers around them, just like we plan to migrate the already existing pickers to those new components.

In summary the `DateTimeRangePicker` will come in two steps:

1. It's keyboard-only component
2. It's full picker with the calendar views

**Comment by bvivek20** [comment]
Upvote

**Comment by Takueg** [comment]
Upvote

**Comment by santoshD93** [comment]
upvote👍

**Comment by rsmml** [comment]
upvote 👍 🚀

**Comment by Aphanite** [comment]
Upvote

**Comment by flaviendelangle** [comment]
As mentioned a few days ago, we now have a new component for date time range editing called `MultiInputDateTimeRangeField` (very long name, but a `SingleInputDateTimeRangeField` is in the pipeline).

The documentation is available [here](https://next.mui.com/x/react-date-pickers/date-time-range-field/)
It is still very early-stage but will serve as the baseline for the future `DateTimeRangePicker`.
So the more feedback we have to improve it, the easier it will be to build this new picker.

**Comment by treeslikemedia** [comment]
Upvote!

**Comment by akdeb** [comment]
This feature would be great to have. My use case revolves around using Cal.com for now but it is paid. To have a MUI Date and time picker would be clutch

**Comment by flaviendelangle** [comment]
To avoid future disappointment, I have to remind that all the range pickers are part of the pro plan (more information on our [pricing page](https://mui.com/pricing/)).

The new components `DateTimeRangePicker` and `TimeRangePicker` will be added to this pro plan.

**Comment by rppala3** [comment]
upvote +1 rocket

**Comment by leonardoviada** [comment]
upvote! 👍

**Comment by DanielJerrehian** [comment]
Upvote 👍

**Comment by kellyndassler** [comment]
Upvote!!

**Comment by grace-fiddler** [comment]
@flaviendelangle when will DateTimeRangePicker and TimeRangePicker available to pro users?

**Comment by flaviendelangle** [comment]
The current goal is to have a first version in Q1 2023.
Note that this component will be released under the v6 version only because it will take advantage of the new internals we use to build the pickers.

**Comment by snooze92** [comment]
Upvote 👍

**Comment by vuongducdai** [comment]
Upvote

**Comment by betterpro** [comment]
Upvote

**Comment by bowann** [comment]
Upvote

**Comment by staceyviolet** [comment]
upvote 👍

**Comment by kobibasson** [comment]
Upvote👍

**Comment by sibsgit** [comment]
Upvote 👍🏽

**Comment by DvirMalka** [comment]
upvote +1

**Comment by darpankattel** [comment]
upvote +1

**Comment by cvcl12345** [comment]
upvote +1

**Comment by marcuxo** [comment]
upvote +1 👍🏼

**Comment by AmitLevinsky** [comment]
upvote +1 👍🏼

**Comment by YanaTrifonova** [comment]
Upvote +1 👍🏼

**Comment by motah-fard** [comment]
upvote +1 👍

**Comment by olgaosinceva** [comment]
upvote! 👍

**Comment by ronald-cognite** [comment]
Upvote! 🚀

**Comment by fondalashay** [comment]
+1 upvote 👍

**Comment by rvillette** [comment]
Upvote!

**Comment by whhc** [comment]
Upvote +1 👍

**Comment by LucHighwalker** [comment]
Upvote! 🤓

**Comment by antonlin1** [comment]
upvote

**Comment by ar-martynov** [comment]
upvote

**Comment by dizag** [comment]
upvote

**Comment by 1ed** [comment]
Could you please upvote in silence? 

There are emojis for that on the original post:

![image](https://user-images.githubusercontent.com/162986/205263352-fa8de935-3447-40a3-b355-754a91af395b.png)

If you want to subscribe for updates you can do that here:

![image](https://user-images.githubusercontent.com/162986/205263461-651ca2ef-aad9-4bc6-9c7a-46b831dbe810.png)

I'm not interested in notifications like "upvote" every day, and maybe I'm not the only one who doesn't. Thanks!

**Comment by amirhatem** [comment]
upvote +1

**Comment by PFJoey** [comment]
+1

**Comment by kamilszewczyk0** [comment]
is there any chance than we can expect this component any time soon?

**Comment by LukasTy** [comment]
> is there any chance than we can expect this component any time soon?

Depends what soon™ means for you. I doubt that we'll finish it before the v6 stable release.
More precise answer—I wouldn't hope for it before Q2. 😉

**Comment by kamilszewczyk0** [comment]
> Depends on what soon™ means for you. I doubt that we'll finish it before the v6 stable release. More precise answer—I wouldn't hope for it before Q2. 😉

In this particular case soon means less than a month 😄 yet appreciate the answer, helped to make a decision 😉

**Comment by rdori** [comment]
+1

**Comment by chwu2004** [comment]
upvote +1 :pray:

**Comment by wave-m** [comment]
I agree.

**Comment by erischon** [comment]
upvote +1

**Comment by vogue-dev** [comment]
upvote +1

**Comment by harshmittal1750** [comment]
wen

**Comment by adarsh-drishya** [comment]
yes when

**Comment by standbyoneself** [comment]
Hi there, when this feature will upcome?

**Comment by LukasTy** [comment]
> Hi there, when this feature will upcome?

Hi. Firstly we need to finish the work on https://github.com/mui/mui-x/issues/4483. After that is done, we will work on the `TimeRangePicker` and `DateTimeRangePicker` components. 👌

**Comment by Gougerik** [comment]
upvote +1

**Comment by OceanTechnic** [comment]
Upvote! +1 👍🏼

**Comment by johnnytolengo** [comment]
Upvote! +1 👍🏼

**Comment by scottdialo** [comment]
Please release this, need it :)

**Comment by PixelWinnerUA** [comment]
Upvote

**Comment by jasminediaconu** [comment]
Upvoting 👍

**Comment by rfrankspk** [comment]
+1

**Comment by GHasanKaraman** [comment]
MUI would be better with this feature for certain.

**Comment by mrigankdoshy** [comment]
> > Hi there, when this feature will upcome?
> 
> Hi. Firstly we need to finish the work on #4483. After that is done, we will work on the `TimeRangePicker` and `DateTimeRangePicker` components. 👌

@LukasTy We're planning to use MUI in a large app in production and would greatly benefit from a Date Time Range Picker component. I see there's a PR up (#9528) , could you please give us a status update or an estimate on when to expect it to land and be available for use?

**Comment by LukasTy** [comment]
@mrigankdoshy We plan to release this component in the coming months. If there will be no major hiccups, maybe even within a month.
But do note, that it will be released only in `v7 alpha`, which is planned to start next week. 😉

**Comment by stoberov** [comment]
@LukasTy - thank you for your continued work on this! :+1: 

Any chance for some update regarding this feature 3 weeks after the last comment?

Sorry to be a pest, only I notice https://github.com/mui/mui-x/pull/9528 has been stuck for 2 weeks with no activity, and the feature has become quite needed at our company :angel:

**Comment by LukasTy** [comment]
@stoberov Thank you for being so interested in this feature. 🙏 
The company had a retreat last week, which stopped some progress. I'll get back to finalizing the PR next week. 😉

**Comment by josephybh** [comment]
@LukasTy Any update? 🙏

**Comment by LukasTy** [comment]
> Any update? 🙏

Thank you for being interested. 🙏 
The PR is awaiting reviews from the team.
I'm splitting certain changes that are necessary and can be delivered beforehand into separate PRs. 😉

**Comment by oliVRqin** [comment]
Updates on this?

**Comment by timur-bwr** [comment]
Hello. 
Do you have any estimations about bringing out this component?

**Comment by joserodolfofreitas** [comment]
Hi everyone, just a heads up: the new Date Time Range Picker is already available on the `next` major version (v7), which is currently in beta.
https://next.mui.com/x/react-date-pickers/date-time-range-picker/

We highly value your feedback and would love to hear your thoughts.
Please feel free to start using it [now](https://next.mui.com/x/migration/migration-pickers-v6/#start-using-the-new-release)!

**Comment by RockPaperGardenHose** [comment]
@joserodolfofreitas Hi! great feature, however will you include the calendar option for this feature as well in the future ? I think ALL the date- and datetimerange pickers should have that option included!

**Comment by RockPaperGardenHose** [comment]
@flaviendelangle 


> [@joserodolfofreitas](https://github.com/joserodolfofreitas) Hi! great feature, however will you include the calendar option for this feature as well in the future ? I think ALL the date- and datetimerange pickers should have that option included!

**Comment by flaviendelangle** [comment]
Hi,

How many months would you like to display at the same time on the `DateTimeRangePicker` component?

**Comment by RockPaperGardenHose** [comment]
Not months on the datetimerangepicker that can ofcourse be whatever the
user chooses but I want to see the datetimerangepicker also have the option
to be a calendar as in like this:
https://mui.com/x/react-date-pickers/date-range-calendar/

**Comment by LukasTy** [comment]
@RockPaperGardenHose Are you referring to the [Static variant](https://mui.com/x/react-date-pickers/date-range-picker/#available-components)?
If that's the case, we have an issue (https://github.com/mui/mui-x/issues/14998) for it you can follow. 😉 

`DateTimeRangePicker` combines the `DateRangeCalendar` and a selection of Digital or Time clock. 😉

**Comment by RockPaperGardenHose** [comment]
Thanks, yes the static variant would be a great addition!
